src/libraries/Common/tests/TestUtilities/System/IO/FileCleanupTestBase.cs

```diff
@@ -214,5 +214,25 @@ private unsafe uint GetFinalPathNameByHandle(SafeFileHandle handle, char[] buffe
                 return Interop.Kernel32.GetFinalPathNameByHandle(handle, bufPtr, (uint)buffer.Length, Interop.Kernel32.FILE_NAME_NORMALIZED);
             }
         }
+
+        protected string CreateTestDirectory(params string[] paths)
+        {
+            string dir = Path.Combine(paths);
+            Directory.CreateDirectory(dir);
+            return dir;
+        }
+
+        protected string CreateTestDirectory()
+             => CreateTestDirectory(GetTestFilePath());
+
+        protected string CreateTestFile(params string[] paths)
+        {
+            string file = Path.Combine(paths);
+            File.Create(file).Dispose();
```

Comment:
```suggestion
            Assert.True(Path.IsPathRooted(file));
            Directory.CreateDirectory(Path.GetDirectoryName(file));
            File.Create(file).Dispose();
```
that way you don't have to create any directory first.

Reply:
I think we will have to create the directories anyways since their paths are used inside most of the methods anyways.

Reply:
Not all of them, eg I pointed out this one
```
            string firstDir = Path.Combine(TestDirectory, "dir1");
            string nestedDir = CreateTestDirectory(firstDir, "nested");
            string nestedFile = CreateTestFile(nestedDir, "nestedFile");
```